contracts/src/libraries/Secp256k1.sol

```diff
@@ -76,22 +120,26 @@ library Secp256k1 {
         return r;
     }
 
-    /// @notice Verifies that `z⋅G - e⋅P = R`.
+    /**
+     * @notice Verifies that `z⋅G - e⋅P = R`.
+     * @param z The scalar multiplier for the generator point.
+     * @param e The scalar multiplier for point P.
+     * @param p The point P.
+     * @param r The expected result point R (witness).
+     * @dev See <https://ethresear.ch/t/you-can-kinda-abuse-ecrecover-to-do-ecmul-in-secp256k1-today/2384>
```

Comment:
nit: The ordering for the NatSpec is not consistent (sometimes, `@dev` comes before the apramters and return, sometimes after). Can we make it consistent.

I would also _love_ to have a `@custom:reference` or such tag for external references like this.


```suggestion
     * @custom:reference https://ethresear.ch/t/you-can-kinda-abuse-ecrecover-to-do-ecmul-in-secp256k1-today/2384
```

We can use it for other links as well.

Reply:
My personal preference is `@dev` at the bottom - like it is here.